dataset/st_dataset.py

```diff
@@ -151,4 +151,21 @@ def concatenate_dataset_dicts(dataset_dicts: List[DatasetDict]) -> DatasetDict:
 # Param: DatasetDict
 # rtype: DatasetInfo 
 def get_dataset_info(data_dict: DatasetDict) -> DatasetInfo:
-    return data_dict["train"].info
\ No newline at end of file
+    return data_dict["train"].info
+
+
+
+def load_dataset_safe(*args, **kwargs) -> Dataset:
+
+    tries = 3
```

Comment:
you could put this in the function parameter list and have a default value. Remember, having a magic number inside of your code is usually not a good idea ;)